test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -74,3 +76,71 @@ def test_replicated_tensor_inter_op_tensor(self):
         self.assertNotIsInstance(new_tensor, ReplicatedTensor)
 
         self.assertEqual(new_tensor, local_tensor + local_rand_tensor)
+
+    @with_comms(init_rpc=False)
+    @skip_if_lt_x_gpu(4)
+    @requires_nccl()
+    def test_with_ddp(self):
+        # Test Replicated params for DDP
+        replica_tensor = ReplicatedTensor(torch.rand(4, 8, device=self.rank))
+        model = torch.nn.Linear(8, 2).cuda(self.rank)
+        original_params = list(model.parameters())
+        ddp = DDP(model)
+        self.assertIsInstance(ddp.module.weight, ReplicatedParameter)
+        self.assertIsInstance(ddp.module.bias, ReplicatedParameter)
+        self.assertIsInstance(model.weight, ReplicatedParameter)
+        self.assertIsInstance(model.bias, ReplicatedParameter)
+
+        # Test module.parameters.
+        params = list(ddp.parameters())
+        self.assertEqual(2, len(params))
+        self.assertEqual(ddp.module.weight, params[0])
+        self.assertEqual(ddp.module.bias, params[1])
+
+        params = list(model.parameters())
+        self.assertEqual(2, len(params))
+        self.assertEqual(model.weight, params[0])
+        self.assertEqual(model.bias, params[1])
```

Comment:
should we add `self.assertTrue(all(isinstance(replicatedParam, param) for param in model.parameters())`?